scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -1326,3 +1326,43 @@ def f(x):
         with pytest.raises(ValueError, match='One of the integrality'):
             DifferentialEvolutionSolver(f, bounds=bounds, polish=False,
                                         integrality=integrality)
+
+    def test_vectorized(self):
```

Comment:
Tests look good. Looks like you checked all the basic functionality and the warnings/errors.

Since I'm not so familiar with the code, I have a hard time being sure that `vectorized` won't interfere with other options. Were you already working with the population as an array of shape `(M, len(x))` throughout the code? Objective function values were already stored in an array of shape `(M,)`, perhaps, so there is very little difference with `vectorized` except in how the function gets evaluated? 

Anyway, I suppose that's why I thought `integrality` and `x0` might not interact well with it. It turns out to be fine, but it might be worth testing that everything is OK with `integrality` and `x0`.

Reply:
> Were you already working with the population as an array of shape (M, len(x)) throughout the code? Objective function values were already stored in an array of shape (M,), perhaps, so there is very little difference with vectorized except in how the function gets evaluated?

Yes, everything is stored as `(M, ...)`. I've jigged the code around so that the objective function gets called with `x.shape==(N, M)`, where N is the number of parameters. That probably makes more sense from a user POV.